frontend/src/utils/date.test.ts

```diff
@@ -0,0 +1,166 @@
+
+import * as DateUtils from './date'
+
+describe('date', () => {
+  describe('isDateBeforeToday', () => {
+    it('should return true when the input date is in the past', () => {
+      const result = DateUtils.isDateBeforeToday(new Date('2023-04-23'))
+
+      expect(result).toBe(true)
+    })
+    it('should return true when the input date is yesterday', () => {
+      const now = new Date()
+      const result = DateUtils.isDateBeforeToday(
+        new Date(now.getFullYear(), now.getMonth(), now.getDate() - 1),
+      )
+
+      expect(result).toBe(true)
+    })
+    it('should return false when the input date is within today (e.g. now)', () => {
+      const result = DateUtils.isDateBeforeToday(new Date())
+
+      expect(result).toBe(false)
+    })
+    it('should return false when the input date is today', () => {
+      const now = new Date()
+      const result = DateUtils.isDateBeforeToday(
+        new Date(now.getFullYear(), now.getMonth(), now.getDate()),
+      )
+
+      expect(result).toBe(false)
+    })
+    it('should return false when the input date is in the future', () => {
+      const now = new Date()
+      const result = DateUtils.isDateBeforeToday(
+        new Date(now.getFullYear(), now.getMonth(), now.getDate() + 1),
+      )
+
+      expect(result).toBe(false)
+    })
+  })
+
+  describe('isDateAfterToday', () => {
+    it('should return true when the input date is in the future', () => {
+      const result = DateUtils.isDateAfterToday(new Date('3023-04-23'))
+
+      expect(result).toBe(true)
+    })
+    it('should return true when the input date is tomorrow', () => {
+      const now = new Date()
+      const result = DateUtils.isDateAfterToday(
+        new Date(now.getFullYear(), now.getMonth(), now.getDate() + 1),
+      )
+
+      expect(result).toBe(true)
+    })
+    it('should return false when the input date is within today (e.g. now)', () => {
+      const result = DateUtils.isDateAfterToday(new Date())
+
+      expect(result).toBe(false)
+    })
+    it('should return false when the input date is today', () => {
+      const now = new Date()
+      const result = DateUtils.isDateAfterToday(
+        new Date(now.getFullYear(), now.getMonth(), now.getDate()),
+      )
+
+      expect(result).toBe(false)
+    })
+    it('should return false when the input date is in the past', () => {
+      const now = new Date()
+      const result = DateUtils.isDateAfterToday(
+        new Date(now.getFullYear(), now.getMonth(), now.getDate() - 1),
+      )
+
+      expect(result).toBe(false)
+    })
+  })
+
+  describe('normalizeDateToUtc', () => {
+    beforeEach(() => {
+      timezoneMock.unregister()
+    })
+    it('should convert local dates to UTC', () => {
+      const dateString = '2023-04-23T00:00:00'
+      const localDate = new Date(Date.parse(dateString))
+      const utcDate = new Date(Date.parse(`${dateString}+00:00`))
+
+      const result = DateUtils.normalizeDateToUtc(localDate)
```

Comment:
Might want to assert before and after the util is called to ensure dates are different first

Reply:
Hm I don't think this is especially valuable as an assertion as ideally the test should still pass when the local time is UTC. But I get it that there is no value in the test if the local time and UTC time is the same, I guess. Will it be better that this test is removed or that the local time here is set as SGT?

Reply:
The rationale is to assert that the mock date library is also working, just in case :P

Reply:
haha okay, I'll add the assertion for the positive/negative utc tests then! yea when i tried the library I did print the dates to make sure hahaha

Reply:
checks added here: [d232fad](https://github.com/opengovsg/FormSG/pull/6261/commits/d232fad80731bd43a00af5cd640898846fc69706)